services/slam/builtin/builtin.go

```diff
@@ -343,11 +343,27 @@ func configureCameras(ctx context.Context, svcConfig *AttrConfig, deps registry.
 			if !ok {
 				return "", nil, transform.NewNoIntrinsicsError("Intrinsics do not exist")
 			}
+
 			err = intrinsics.CheckValid()
 			if err != nil {
 				return "", nil, err
 			}
+
+			props, err := cam.Properties(ctx)
+			if err != nil {
+				return "", nil, errors.Wrapf(err, "error getting props %v for slam service")
+			}
+
+			distortion, ok := props.DistortionParams.(*transform.BrownConrady)
+			if !ok {
+				return "", nil, errors.New("error getting distortion_parameters for slam service")
```

Comment:
I think you can also explicitly say that only BrownConrady distortion parameters are supported.